fiboa_cli/util.py

```diff
@@ -105,6 +105,8 @@ def get_fs(url_or_path: str) -> AbstractFileSystem:
     parsed = urlparse(url_or_path)
 
     if parsed.scheme in ('http', 'https'):
+        if re.search(r"[?&]blocksize=0", url_or_path):
+            return HTTPFileSystem(block_size=0)
```

Comment:
That looks suspicious. What's this? :-)

Reply:
I agree. However the program fails without this option, and the error suggests adding this `blocksize=0` option. We're using an `fsspec.HTTPFileSystem` and download the file chunked in `convert_utils.py:stream_file()`. This is implemented by doing http-range requests of 10M size, but the origin server for this dataset does not support http-range requests and fails.

This was the shortest path to introducing this option.

https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.implementations.http.HTTPFileSystem

Reply:
Yes, I tried: https://github.com/fiboa/cli/pull/95/files#diff-73f13a18826c38c2ef4d83a2a5252675d1b156fad00c4f39b3cd9b51d44e75f7R108

All source datasets slightly differ. It leads us to a `convert_utils.py:convert()` function with 25 arguments + kwargs, and it still is not enough to capture all variations. I muffled this `blocksize` argument into the url to avoid introducing more. If we would refactor everything to a class-based Converter approach (with a BaseConverter, maybe some mixins, for each country a derrived Converter class), we could override a `self.get_fs()` and handle local variations locally...

Reply:
Yeah, we are hitting limits here ;-)

Thanks for adding the comment.